appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartData.java

```diff
@@ -0,0 +1,137 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.gwt.resources.client.ImageResource;
+import com.google.gwt.user.client.ui.Image;
+import com.google.gwt.user.client.ui.InlineHTML;
+import com.google.gwt.user.client.ui.SimplePanel;
+
+public abstract class MockChartData extends MockVisibleComponent {
```

Comment:
Both the Android and the GWT implementations comprise of the following classes:
* Chart (central component with type choices)
* ChartData (component that can be dragged onto a Chart)
* ChartDataModel (class responsible for handling data logic and property changes, not visible to the user)
* ChartView (class responsible for handling the Chart, initialization of Data, and Chart-specific properties)

This follows the design described in the document. Should the design be altered in some way? Generally, I found it to work really well in practice, allowing to separate the concerns as much as possible, and adding some transparency to the components themselves.

The main reason for this separation is because Chart & ChartData components are broader than the actual VIews and Models, which then handle each Chart case individually (e.g. View and Model classes for Scatter, Line and Pie Charts are separate classes).

Reply:
For reference, the following design is implemented:

![Chart Type-based Design](https://user-images.githubusercontent.com/34982762/60893108-97de5000-a260-11e9-8b4a-92ffb120b433.png)